File: road-trip-frontend/src/util/UserService.js
import axios from 'axios';
import http from "../http-common"
const USER_REST_API_URL = 'http://localhost:8080/user'

class UserService{
    getAll(){
        return http.get("/user");
    }
    get(id){
        return http.get(`/user/${id}`);
    }
    create(data){
        return http.post("user", data);
    }
    update(id, data){
        return http.put(`/user/${id}`, data);
    }
    delete(id){
        return http.delete(`/user/${id}`);
    }
    deleteAll(){
        return http.delete(`/user`);
    }
    findByEmail(email){
        return http.get(`user?email=${email}`);
    }
}
export default new UserService();

File: road-trip-api/src/main/java/road/trip/api/user/UserService.java
package road.trip.api.user;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;

@Service
public class UserService {

    @Autowired
    public UserRepository userRepository;

    public Optional<User> findUser(Long userId) {
        return userRepository.findById(userId);
    }
    //public Optional<User> findByEmail(String email){ return userRepository.findByEmailAddress(email); }
    public Optional<User> findUserByEmail(String email){
        return userRepository.findByUsername(email);
    }

    
    public void deleteUser(Long id){ userRepository.deleteById(id); }
    @Transactional
    public void deleteAllUsers() { userRepository.deleteAll(); }

    @Transactional
    //Create User
    public User saveUser(User user) {
        return userRepository.save(user);
    }

    public List<User> getAllUsers() { return userRepository.findAll(); }


}
